lib/openshift/machine_set.rb

```diff
@@ -7,16 +7,16 @@ class MachineSet < ProjectResource
 
     def desired_replicas(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
-      rr.dig('spec', 'replicas')
+      rr.dig('spec', 'replicas').to_i
     end
 
-    def current_replicas(user: nil, cached: true, quiet: false)
+    def available_replicas(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
-      rr.dig('status', 'replicas')
+      rr.dig('status', 'availableReplicas').to_i
     end
 
     def healthy?
-      return desired_replicas == current_replicas
+      return desired_replicas == available_replicas
```

Comment:
actually it makes more sense to add some params:
```
    def healthy?(user: nil, cached: true, quiet: false)
      return desired_replicas(user: user, cached: cached, quiet: quiet) == available_replicas(cached: true)
    end
```
Otherwise it might be ugly to call the method in a wait loop.

Reply:
I would suggest also to rename this method to `ready?`. Then waiting for this will be as easy as calling
```
  machine_set.wait_till_ready(user, timeout)
```